### PR TITLE
[CLI][Projects] Save the project by default if not exist when loading/running a project through the CLI

### PR DESCRIPTION
**Pull request by Eyal-Danieli**
Until now, the user has been required to pass `--ensure-project` when loading or running a project through the CLI to create a project if not exist. In this PR, we create the project by default. 
Moreover, the `--ensure-project` flag will be deprecated (and will be removed in `1.5.0`) and instead we will be using a new flag called `save` to be consistent with the corresponding `save` parameter within `mlrun.projects.project.load_project()`. The new `save` flag is `True` by default.

**UPDATE**
In this PR we also enrich the git source url during the load project process if it has no branch and refs. e.g. `git://github.com/mlrun/project-demo.git` will be converted to `git://github.com/mlrun/project-demo.git#refs/heads/main`(assuming that `main` is the main active branch of the repo).

### CODE REVIEW

File: mlrun/__main__.py [hunk 1]
```diff
@@ -1006,11 +1007,18 @@ def logs(uid, project, offset, db, watch):
...
     "--ensure-project",
     is_flag=True,
     help="ensure the project exists, if not, create project",
 )
+@click.option(
+    "--save",
+    is_flag=True,
+    default=True,
+    help="create and save the project if not exist",
+)
```
Comment (Tankilevitch) [inline mlrun/__main__.py:1021]: if default is True (when not passing) and this is a flag, so also passing it `--save` will be True, so I think you'll never reach False.

Comment (Tankilevitch) [reply mlrun/__main__.py:1021]: take a look at - https://click.palletsprojects.com/en/8.1.x/options/#boolean-flags 
guess we can allow both `save and no-save` in the same argument with default true

File: mlrun/__main__.py [hunk 2]
```diff
@@ -1062,17 +1070,23 @@ def project(
...
     """load and/or run a project"""
     if env_file:
         mlrun.set_env_from_file(env_file)
 
+    if ensure_project:
+        warnings.warn(
+            "'ensure_project' is deprecated and will be removed in 1.6.0, use 'save' (True by default) instead. ",
+            # TODO: Remove this in 1.6.0
+            FutureWarning,
+        )
```
Comment (Tankilevitch) [inline mlrun/__main__.py:1084]: I think there might be an option to add those deprecation warning in `click`

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
Few changes are still needed, also lets add a test covering that new option, I guess in the system tests is alright

File: mlrun/projects/project.py [hunk 3]
```diff
@@ -868,6 +891,16 @@ def set_source(self, source, pull_at_runtime=False, workdir=None):
         """
         self.spec.load_source_on_run = pull_at_runtime
         self.spec.source = source or self.spec.source
+
+        if self.spec.source.startswith("git://"):
+
+            source, reference, branch = resolve_git_reference_from_source(source)
+            if not branch and not reference:
+                warnings.warn(
```
Comment (Tankilevitch) [inline mlrun/projects/project.py:899]: is there a reason not to use our logger?

Comment (Tankilevitch) [reply mlrun/projects/project.py:899]: does it reproduce different output when using `warnings.warn` vs `logger.warning`?

File: tests/system/projects/test_project.py [hunk 4]
```diff
@@ -315,6 +313,27 @@ def test_inline_pipeline(self):
...
+        project_dir = f"{projects_dir}/{name}"
+        shutil.rmtree(project_dir, ignore_errors=True)
+
+        # clone a project to local dir but don't save the project to the DB
+        args = [
+            "-n",
+            name,
+            "-u",
+            "git://github.com/mlrun/project-demo.git",
+            "--no-save",
```
Comment (Tankilevitch) [inline tests/system/projects/test_project.py:329]: I now see that u made this test without specifying any branch or ref, but worth having one as well in the test_projects / test_archives

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
looks good, few comments

File: mlrun/projects/project.py (hunk 3 above)
Comment (Eyal-Danieli) [reply mlrun/projects/project.py:899]: right, `warnings.warn` is more for deprecation warnings while the `logger.warn` is more for a specific warning for the user. In our case is should be `logger.warn`.

File: mlrun/utils/helpers.py [hunk 5]
```diff
@@ -1256,3 +1257,20 @@ def resolve_git_reference_from_source(source):
         return split_source[0], reference, ""
 
     return split_source[0], "", reference
+
+
+def ensure_git_branch(url: str, repo: git.Repo) -> str:
+    """Ensures git url includes branch. If no branch or refs are included in the git source then will enrich the
```
Comment (Tankilevitch) [inline mlrun/utils/helpers.py:1263]:
```suggestion
    """Ensures git url includes branch. 
    If no branch or refs are included in the git source then will enrich the
```

**Review by Tankilevitch — APPROVED** [review]
Looks good, minor thing that annoyed me, please change it and will merge

**Pull request merged**